www/js/pedagogical_module.js

```diff
@@ -0,0 +1,217 @@
+
+		getActionForType: function(id, answer){
+			/*
+			* This function gives node properties to be shown as per the values
+			* entered by the user for the node.
+			* This is called when the user reopens a node.
+			*/
+			var nodePart = "variableType";
+			var interpretation = this._getInterpretation(id, nodePart, answer);
+			var obj = [];
+			var returnObj = [];
+			
+			nodeEditorActionTable[interpretation][this.userType](returnObj, nodePart);
+			
+			console.log("Return Obj" , returnObj);
+			return returnObj;
+		},
+
+		processAnswer: function(/*string*/ id, /*string*/ nodePart, /*string | object*/ answer,/*string*/ answerString){
+			// Summary: Pocesses a student's answers and returns if correct, 
+			//		incorrect, etc. and alerts the controller about what parts 
+			//		of the node editor should be active.
+			//
+			// Tags: Private
+			var actual_id = this.model.student.getAuthoredID(id);
+			console.log("actual id is",actual_id);
+			var interpretation = this._getInterpretation(id, nodePart, answer);
+			
+			var returnObj = [], currentStatus;
+			var givenID ;  // ID of the correct node, if it exists
+			var solutionGiven = false;
+			var givenAnswer = answer; //keeping a copy of answer for logging purposes.
+			
+			var updateStatus = function(returnObj, model){
+				
+				returnObj.forEach(function(i){
+					if(i.attribute === "status"){
+						if(i.value === "correct"){
+							if(model.authored.getStatus(givenID, nodePart) !== "demo"){
+								model.authored.setStatus(givenID, nodePart, "correct");
+							}
+						}else{
+							model.authored.setStatus(givenID, nodePart, "incorrect");
+						}
+						model.authored.getStatus(givenID, nodePart)
+					}
+				});
+			};
+			if(answer){
+				if(nodePart === "description"){
+					givenID = answer;
+					descriptionTable[interpretation][this.userType](returnObj, nodePart);
+					for(var i = 0; i < returnObj.length; i++){
+						if(returnObj[i].value === "correct"){
+							currentStatus = this.model.authored.getStatus(givenID, nodePart); //get current status set in given model
+							// Set failure attempts based on current status
+							if(currentStatus === "")
+								this.model.authored.setStatus(givenID, nodePart, returnObj[i].value);
+							else
+								updateStatus(returnObj, this.model);
+							this.descriptionCounter = 0;
+						}
+					}
+				}else{
+					givenID = this.model.student.getAuthoredID(id);
+					console.assert(nodeEditorActionTable[interpretation], "processAnswer() interpretation '" + interpretation + "' not in table ", nodeEditorActionTable);
+					nodeEditorActionTable[interpretation][this.userType](returnObj, nodePart);
+					
+					currentStatus = this.model.authored.getStatus(givenID, nodePart); //get current status set in given model
+					if (currentStatus !== "correct") {
+						for (var i = 0; i < returnObj.length; i++)
+							if (returnObj[i].value === "incorrect") {
+								// handle incorrect result - increment assistance score
```

Comment:
I am just going through this commit by commit. I have totally forgotten whether assisstance score is being kept or not. I hope it is.